pose-detection/demos/live_video/src/index.js

```diff
@@ -124,17 +128,19 @@ async function renderResult() {
   // The null check makes sure the UI is not in the middle of changing to a
   // different model. If during model change, the result is from an old model,
   // which shouldn't be rendered.
-  if (poses.length > 0 && !STATE.isModelChanged) {
+  if (poses.length > 0) {
     camera.drawResults(poses);
   }
 }
 
 async function renderPrediction() {
   await checkGuiUpdate();
 
-  await renderResult();
+  if (!STATE.isModelChanged) {
+    await renderResult();
+  }
 
-  requestAnimationFrame(renderPrediction);
+  rafId = requestAnimationFrame(renderPrediction);
```

Comment:
Just curious: one thing I don't fully understand is how this rafId can still be used when you use it in checkGuiUpdate(). Isn't checkGuiUpdate called from the renderPrediction() that is associated with that rafId?